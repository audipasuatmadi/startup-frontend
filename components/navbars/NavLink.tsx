import React from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';

interface Props {
  href: string;
  children: JSX.Element;
  activeClassName: string;
  exact?: boolean
}

const NavLink = ({ href, children, activeClassName, exact }: Props) => {
  const router = useRouter();

  const routePath = exact? router.pathname : router.pathname.split('/')[1]
  const className =
    routePath === href.split('/')[1]
      ? `${children.props.className} ${activeClassName}`
      : `${children.props.className}`;

  return <Link href={href}>{React.cloneElement(children, { className })}</Link>;
};

export default NavLink;
