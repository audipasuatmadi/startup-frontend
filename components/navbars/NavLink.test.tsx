import React from 'react';
import { shallow } from 'enzyme';
import NavLink from './NavLink';
import { useRouter } from 'next/router';

jest.mock('next/router');

describe('NavLink rendering tests', () => {
  let mockedUseRouter: jest.Mock;

  beforeEach(() => {
    mockedUseRouter = useRouter as jest.Mock;
  });

  afterEach(() => {
    mockedUseRouter.mockClear();
  });

  it('should render correctly if current route does not match', () => {
    mockedUseRouter.mockReturnValue({ pathname: '/aboutus' });

    const wrapper = shallow(
      <NavLink href='/home' activeClassName='bg-red-900'>
        <a className='test'>Dummy Children</a>
      </NavLink>
    );

    expect(mockedUseRouter).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly if current route does match', () => {
    mockedUseRouter.mockReturnValue({ pathname: '/home' });

    const wrapper = shallow(
      <NavLink href='/home' activeClassName='bg-red-900'>
        <a className='test '>Dummy Children</a>
      </NavLink>
    );

    expect(mockedUseRouter).toHaveBeenCalled();
    expect(wrapper.find('a').prop('className').split(' ')[2]).toMatch(
      'bg-red-900'
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correcly if current route is a children of the main route', () => {
    mockedUseRouter.mockReturnValue({ pathname: '/home/subroute' });

    const wrapper = shallow(
      <NavLink href='/home' activeClassName='bg-red-900'>
        <a className='test '>Dummy Children</a>
      </NavLink>
    );

    expect(mockedUseRouter).toHaveBeenCalled();
    expect(wrapper.find('a').prop('className').split(' ')[2]).toMatch(
      'bg-red-900'
    );
    expect(wrapper).toMatchSnapshot();
  });
});
