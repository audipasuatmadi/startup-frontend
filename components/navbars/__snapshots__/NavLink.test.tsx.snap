// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink rendering tests should render correcly if current route is a children of the main route 1`] = `
<Link
  href="/home"
>
  <a
    className="test  bg-red-900"
  >
    Dummy Children
  </a>
</Link>
`;

exports[`NavLink rendering tests should render correctly if current route does match 1`] = `
<Link
  href="/home"
>
  <a
    className="test  bg-red-900"
  >
    Dummy Children
  </a>
</Link>
`;

exports[`NavLink rendering tests should render correctly if current route does not match 1`] = `
<Link
  href="/home"
>
  <a
    className="test"
  >
    Dummy Children
  </a>
</Link>
`;
